assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/multiuser.properties

```diff
@@ -40,6 +40,10 @@ che.limits.workspace.env.ram=16gb
 # counts toward idleness.
 che.limits.workspace.idle.timeout=1800000
 
+# The length of time that a workspace will run, regardless of activity, before
+# the system will suspend it.
+che.limits.workspace.run.timeout=10800000
```

Comment:
1. If I'm not mistaken `che.limits.workspace.idle.timeout` is just a default value applied to every user, but then admin is able to override it per user. Does it make sense to implement the same for run.timeout?
2. From UX perspective I suppose it would be nice to see a warning when like there are 3 minutes left before workspace is shutdown...

Reply:
++

Reply:
> From UX perspective I suppose it would be nice to see a warning when like there are 3 minutes left before workspace is shutdown...

Is it correct that this would require a che-theia change?

Reply:
> Is it correct that this would require a che-theia change?

I suppose - yes. But as discussed with Illya - the only thing that makes sense to do with that in the current PR - create an issue for CheServer/Dashboard/Che Theia to improve UX when workspace is stopped by run_timeout or idle_timeout.